src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -222,36 +182,47 @@
     b
     (Boolean/parseBoolean b)))
 
-(defn count?
-  "If the original query contains the include_total w/o a a value, it
-  will show up here as nil.  We assume that in that case, the caller
-  intended to use it as a flag"
-  [query-map]
-  (let [total (get query-map :include_total ::not-found)]
-    (if (= total ::not-found)
-      false
-      (coerce-to-boolean total))))
+(defn validate-query-params
+  "Given a set of params and a param spec, throw and error if required params
+   are missing or unsupported params are present, otherwise return the params."
+  [params param-spec]
+  (let [params (stringify-keys params)]
+    (kitchensink/cond-let
```

Comment:
can't say I'm a huge fan of cond-let, but if it simplifies the code I guess it's good.